File: LC-NOTES/src/main/java/iu/projects/lcnotes/LambdaHandler.java
package iu.projects.lcnotes;

import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.events.ScheduledEvent;

public class LambdaHandler implements RequestHandler<ScheduledEvent, String> {

	/*
	public String myHandler(int anInt) {
        return Main.sendEmail();
    }
    */
	
	public String handleRequest(ScheduledEvent event, Context context) {
		LambdaHandlerScala$.MODULE$.sendEmail();
		return String.format("Email sent %s", event.getTime().toString());
	}
}

File: aws-lambda/src/main/scala/urisman/wp/ping/LambdaHandler.scala
package urisman.wp.ping

import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
import com.amazonaws.services.lambda.runtime.events.ScheduledEvent

import java.net.{HttpURLConnection, URL}
import scala.io.Source

class LambdaHandler extends RequestHandler[ScheduledEvent, String] {
	
	private val to = "igor.urisman@gmail.com"
	private def subject(status: Int): String = status match {
		case 200 => "urisman.net: OK"
		case _@code => s"*** urisman.net: $code ***"
	}

	def handleRequest(event: ScheduledEvent, context: Context): String = {
		val url = new URL("https://www.urisman.net/")
		val conn = url.openConnection().asInstanceOf[HttpURLConnection]
		conn.setRequestMethod("GET")
		val subj = subject(conn.getResponseCode)
		Email.sendText(to = to, from = to, subject = subj, body = "")
		context.getLogger.log(subj)
		subj
	}
}
